ui/apps/platform/src/Components/CompoundSearchFilter/types.ts

```diff
@@ -5,6 +5,12 @@ import { SearchCategory } from 'services/SearchService';
 
 export type BaseInputType = 'autocomplete' | 'text' | 'date-picker' | 'condition-number';
 export type InputType = BaseInputType | 'select';
+export type SelectSearchFilterOptions = {
+    options: { label: string; value: string }[];
+};
+export type SelectSearchFilterGroupedOptions = {
+    groupOptions: { name: string; options: { label: string; value: string }[] }[];
+};
```

Comment:
Nit: When rendering the CompoundSearchFilterInput component, if the incoming options object has both `options` and `groupOptions` properties, it will silently ignore the singular version and always render the grouped version.

Here are a couple of ideas on how we could model this to make that case less likely, but nothing worth changing in this PR.

```ts
// Extract this common shape for reuse/keeping types in sync
export type SelectSearchFilterOptions = {
    options: { label: string; value: string }[];
};

// Add a `type` discriminator field that can be used to determine if the options are grouped or not
export type SelectSearchFilterSingularOptions = {
    type: 'singular';
} & SelectSearchFilterOptions;

export type SelectSearchFilterGroupedOptions = {
    type: 'grouped';
    groupOptions: { name: string; } & SelectSearchFilterOptions [];
};
```

Or another option, which is easier to read and work with, but is less restrictive than the above:
```ts
export type SelectSearchFilterOptions = {
    options: { label: string; value: string }[];
    groupOptions?: never;
};
export type SelectSearchFilterGroupedOptions = {
    options?: never;
    groupOptions: { name: string; options: { label: string; value: string }[] }[];
};
```

Reply:
I like the thought process. I didn't consider someone adding both. Although it should be unlikely someone would use both, it's something we can restrict. I'm more in favor of the first approach. Maybe we can consider this in a follow-up PR.